src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpRequestParser.cs

```diff
@@ -47,6 +51,88 @@ internal class DefaultHttpRequestParser : IHttpRequestParser
             CancellationToken cancellationToken) =>
             ReadAsync(stream, false, cancellationToken);
 
+        public async ValueTask<IReadOnlyList<GraphQLRequest>> ReadMultipartRequestAsync(
+            HttpRequest httpRequest,
+            CancellationToken cancellationToken)
+        {
+            // todo: The IFormCollection is convenient, but it requires us to work with strings instead of a stream
+            var formFeature = new FormFeature(httpRequest);
+            var form = await formFeature.ReadFormAsync(cancellationToken);
+
+            if (form.Count < 2)
+                throw new Exception("At least a 'operations' and 'map' field need to be present.");
+
+            if (!form.TryGetValue("operations", out StringValues operationsCollection) || !operationsCollection.TryPeek(out var operationsString))
+                throw new Exception("No 'operations' specified.");
+
+            if (!form.TryGetValue("map", out StringValues mapCollection) || !mapCollection.TryPeek(out var mapString))
+                throw new Exception("No 'map' specified.");
+
+            var map = JsonSerializer.Deserialize<IDictionary<string, string[]>>(mapString);
+
+            if (map == null)
+                throw new Exception("'map' is not a dictionary of string[].");
+
+            // todo: Stuff like this is just a temporary hack until we have a proper multipart reader
+            var operationsBytes = Encoding.UTF8.GetBytes(operationsString);
+            var operationsStream = new MemoryStream(operationsBytes);
+
+            var requests = await ReadAsync(operationsStream, false, cancellationToken);
+
+            if (requests == null)
+                throw new Exception("Requests could not be parsed.");
+
+            foreach (var request in requests)
+            {
+                // todo: This is quite hacky and requires Utf8GraphQLRequestParser to actually return this
```

Comment:
I do not necessarily think that we have to move this into the parser itself. The parser actually did do all the parsing. This is kind of a post-processing of the variables. We should however give that its own method.